nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/test/java/org/apache/nifi/processors/mongodb/GetMongoTest.java

```diff
@@ -69,6 +70,7 @@
     @Before
     public void setup() {
         runner = TestRunners.newTestRunner(GetMongo.class);
+        runner.setProperty(GetMongo.QUERY, "{}");
```

Comment:
Does this need to be here? Seems like it might hide the cases when Query is empty, which would be the existing behavior. If it's because of the validator, then it should accept an empty query as valid since the property is not required.  Also, can you add a unit test (or update the existing one) for when the Query is empty and there is no incoming flow file (but there is an incoming connection)? That would test the other half of the conditions for the exception block you added at the top of onTrigger.

Reply:
Ok. What I'm going to do there is make it so that empty query is only valid when there is no incoming connection.

Reply:
I don't think you have access to connection information during validation (all you get is a ValidationContext), I was looking into adding it at one point, but never got around to it.